src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java

```diff
@@ -0,0 +1,281 @@
+     */
+    public enum Options {
+
+        /**
+         * Comments nodes should be processed.
+         */
+        WITH_COMMENTS,
+
+        /**
+         * Comments nodes should be ignored.
+         */
+        WITHOUT_COMMENTS
+
+    }
+
+    /** Stop instances being created. **/
+    private JavaParser() {
+    }
+
+    /**
+     * Static helper method to parses a Java source file.
+     * @param contents contains the contents of the file
+     * @return the root of the AST
+     * @throws CheckstyleException if the contents is not a valid Java source
+     */
+    public static DetailAST parse(FileContents contents)
+            throws CheckstyleException {
+        final String fullText = contents.getText().getFullText().toString();
+        final Reader reader = new StringReader(fullText);
+        final GeneratedJavaLexer lexer = new GeneratedJavaLexer(reader);
+        lexer.setCommentListener(contents);
+        lexer.setTokenObjectClass("antlr.CommonHiddenStreamToken");
+
+        final TokenStreamHiddenTokenFilter filter =
+                new TokenStreamHiddenTokenFilter(lexer);
+        filter.hide(TokenTypes.SINGLE_LINE_COMMENT);
+        filter.hide(TokenTypes.BLOCK_COMMENT_BEGIN);
+
+        final GeneratedJavaRecognizer parser =
+            new GeneratedJavaRecognizer(filter);
+        parser.setFilename(contents.getFileName());
+        parser.setASTNodeClass(DetailAST.class.getName());
+        try {
+            parser.compilationUnit();
+        }
+        catch (RecognitionException | TokenStreamException ex) {
+            final String exceptionMsg = String.format(Locale.ROOT,
+                "%s occurred during the analysis of file %s.",
+                ex.getClass().getSimpleName(), contents.getFileName());
+            throw new CheckstyleException(exceptionMsg, ex);
+        }
+
+        return (DetailAST) parser.getAST();
+    }
+
+    /**
+     * Parses Java source file.
+     * @param file the file to parse
+     * @param withComments {@link Options#WITH_COMMENTS} to include comment nodes to the AST
+     * @return DetailAST tree
+     * @throws IOException if the file could not be read
+     * @throws CheckstyleException if the file is not a valid Java source file
+     */
+    public static DetailAST parseFile(File file, Options withComments)
+            throws IOException, CheckstyleException {
+        final FileText text = new FileText(file.getAbsoluteFile(),
+            System.getProperty("file.encoding", StandardCharsets.UTF_8.name()));
+        return parseFileText(text, withComments);
+    }
+
+    /**
+     * Parses Java source file. Result AST does not contains comment nodes.
+     * @param file the file to parse
+     * @return DetailAST tree
+     * @throws IOException if the file could not be read
+     * @throws CheckstyleException if the file is not a valid Java source file
+     */
+    public static DetailAST parseFile(File file)
```

Comment:
I think we should remove this method.
If we leave these in, what is the point of making `parseFile(File file, Options withComments)` `public`? We have a method for both options.

Reply:
done.